src/frontend/flight/handler.rs

```diff
@@ -168,6 +169,12 @@ impl SeafowlFlightHandler {
                 .get_internal_object_store()?
                 .get_log_store(&cmd.path),
             Some(store_loc) => {
+                let mut options = store_loc.options;
+                if store_loc.location.starts_with("file://") {
+                    // In case of a local FS create the table path
+                    options.insert(CREATE_DIR.to_string(), "true".to_string());
+                }
+
                 self.context
                     .metastore
                     .object_stores
```

Comment:
Come to think about it, maybe I'd add it here instead, add a parameter to this function like `get_log_store_for_table(for_writes: bool)` and then avoid having to thread through a magic CREATE_DIR setting and figure out which location it is based on the protocol which feels like a leaky abstraction.

Reply:
Hmm, doing that kinda feels like a leaky abstraction too? 

I would either need to pass `for_writes: bool` down into `object_store_factory::build_object_store_from_opts` too (which seems redundant, all object stores should be writable), or add CREATE_DIR at that level to the optoins based on the protocol/scheme, same as now.

Maybe the best solution is just to go with wrapping `LocalFileSystem::new()` with an appropriate `PrefixStore` then?